JavaScript/JavaScriptSDK/Logging.ts

```diff
@@ -182,7 +186,21 @@
                 if (typeof (message) !== "undefined" && !!message) {
                     if (typeof (message.message) !== "undefined") {
                         message.message = this.AiUserActionablePrefix + message.message;
-                        this.warnToConsole(message.message);
+
+                        // check if this message type was already logged to console for this page view and if so, don't log it again
+                        var logToConsole = true;
```

Comment:
I would remove the variable and invert the if statement below to have the key = true setting, remove the else, and just put the console call inside the conditional for shorter/simpler/faster code.